src/core/load-fragment.c

```diff
@@ -6573,3 +6575,50 @@ int config_parse_tty_size(
 
         return config_parse_unsigned(unit, filename, line, section, section_line, lvalue, ltype, rvalue, data, userdata);
 }
+
+int config_parse_log_filter_regex(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        _cleanup_(pattern_freep) pcre2_code *cs = NULL;
+        ExecContext *c = ASSERT_PTR(data);
+        const char *pattern = ASSERT_PTR(rvalue);
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+
+        if (pattern[0] == '~')
+                /* pattern is not NULL (checked by ASSERT_PTR()), so accessing pattern[] shouldn't be
+                 * an issue. */
+                ++pattern;
+
+        if (isempty(pattern)) {
+                c->log_filter_regex = mfree(c->log_filter_regex);
+                return 0;
+        }
+
```

Comment:
Hmm, this means that if we have `LogFilterRegex=~`, we discard the value, is that the behavior we want? I think we should only clear the existing value if `LogFilterRegex=` is used. And `LogFilterRegex=~` should probably be disallowed altogether (`LogFilterRegex=~.*` should be used instead).